src/node_task_runner.cc

```diff
@@ -269,29 +272,50 @@ void RunTask(std::shared_ptr<InitializationResultImpl> result,
   simdjson::ondemand::object main_object;
   simdjson::error_code error = json_parser.iterate(raw_json).get(document);
 
-  // If document is not an object, throw an error.
-  if (error || document.get_object().get(main_object)) {
-    fprintf(stderr, "Can't parse package.json\n");
+  if (error) {
+    fprintf(stderr, "Can't parse %s\n", path.string().c_str());
     result->exit_code_ = ExitCode::kGenericUserError;
     return;
   }
+  // If document is not an object, throw an error.
+  auto root_error = document.get_object().get(main_object);
+  if (root_error) {
+    if (root_error == simdjson::error_code::INCORRECT_TYPE) {
+      fprintf(stderr,
+              "Root value unexpected not an object for %s\n\n",
+              path.string().c_str());
+    } else {
+      fprintf(stderr, "Can't parse %s\n", path.string().c_str());
+      result->exit_code_ = ExitCode::kGenericUserError;
+      return;
+    }
+  }
 
   // If package_json object doesn't have "scripts" field, throw an error.
   simdjson::ondemand::object scripts_object;
   if (main_object["scripts"].get_object().get(scripts_object)) {
-    fprintf(stderr, "Can't find \"scripts\" field in package.json\n");
+    fprintf(stderr, "Can't find \"scripts\" field in %s\n", path.string().c_str());
```

Comment:
It seems, std::filesystem::path has `.c_str()`

```suggestion
    fprintf(stderr, "Can't find \"scripts\" field in %s\n", path.c_str());
```